generator/views/glean_ping_view.py

```diff
@@ -35,6 +36,66 @@ class GleanPingView(PingView):
     type: str = "glean_ping_view"
     allow_glean: bool = True
 
+    def to_lookml(self, bq_client, v1_name: Optional[str]) -> Dict[str, Any]:
+        """Generate LookML for this view.
+
+        The Glean views include a labeled metrics, which need to be joined
+        against the view in the explore.
+        """
+        lookml = super().to_lookml(bq_client, v1_name)
+
+        # iterate over all of the glean metrics and generate views for unnested
+        # fields as necessary. Append them to the list of existing view
+        # definitions.
+        table = next(
+            (table for table in self.tables if table.get("channel") == "release"),
+            self.tables[0],
+        )["table"]
+        dimensions = self.get_dimensions(bq_client, table, v1_name)
+
+        client_id_field = self._get_client_id(dimensions, table)
+
+        view_definitions = []
+        metrics = self._get_glean_metrics(v1_name)
+        for metric in metrics:
+            if metric.type == "labeled_counter":
+                looker_name = self._to_looker_name(metric)
+                definition = {
+                    "name": f"{self.name}__{looker_name}",
+                    "label": (
+                        "_".join(looker_name.split("__")[1:]).replace("_", " ").title()
+                    ),
+                    "dimensions": [
+                        {"name": "key", "type": "string", "sql": "${TABLE}.key"},
+                        {
+                            "name": "value",
+                            "type": "number",
+                            "sql": "${TABLE}.value",
+                            "hidden": "yes",
+                        },
+                    ],
+                    "measures": [
+                        {
+                            "name": "count",
+                            "type": "sum",
+                            "sql": "${value}",
+                        },
+                        {
+                            "name": "client_count",
+                            "type": "count_distinct",
+                            "sql": f"case when ${{value}} > 0 then ${{{self.name}.{client_id_field}}} end",
+                        },
+                    ],
+                }
+                view_definitions += [definition]
+        # deduplicate view definitions, because somehow a few entries make it in
+        # twice e.g. metrics__metrics__labeled_counter__media_audio_init_failure
```

Comment:
Do you have a sense for why there are duplicate defintions? Is it because there are sub-dimensions for the metric?